### PR TITLE
LPS-104318 Improve filters structure using react hooks

### PR DESCRIPTION
**Pull request by gabrielpramos**

### CODE REVIEW

**Comment by gabrielpramos** [comment]
ci:test:sf

**Comment by gabrielpramos** [comment]
ci:test:relevant

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/filter/SLAStatusFilterHooks.es.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	{
+		key: slaStatusConstants.untracked,
+		name: Liferay.Language.get('untracked')
+	}
+];
+
+const SLAStatusFilter = ({
+	className,
+	dispatch,
+	filterKey = 'slaStatuses',
```
Comment (albinojunior) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/filter/SLAStatusFilterHooks.es.js]: Use constants filter key.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/filter/TimeRangeFilter.es.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import {CustomTimeRangeForm} from './CustomTimeRangeForm.es';
+import {
+	getCustomTimeRangeName,
+	TimeRangeContext
+} from './store/TimeRangeStore.es';
+
+const TimeRangeFilter = ({
+	buttonClassName,
+	className,
+	filterKey = 'timeRange',
```
Comment (albinojunior) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/filter/TimeRangeFilter.es.js]: Use constants filter key.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/filter/TimeRangeFilterHooks.es.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+	return preventDefault;
+};
+
+const TimeRangeFilter = ({
+	className,
+	dateEndKey = 'dateEnd',
+	dateStartKey = 'dateStart',
+	dispatch,
+	filterKey = 'timeRange',
```
Comment (albinojunior) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/filter/TimeRangeFilterHooks.es.js]: Use constants filter key.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/filter/TimeRangeFilterHooks.es.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+	options: {
+		hideControl = false,
+		multiple = false,
+		position = 'left',
+		withSelectionTitle = false
+	} = {}
```
Comment (albinojunior) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/filter/TimeRangeFilterHooks.es.js]: Define options property with a default value and remove destruction and passing options "{...options}" to Filter component.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/filter/VelocityUnitFilterHooks.es.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+const VelocityUnitFilter = ({
+	className,
+	dispatch,
+	filterKey = 'velocityUnit',
+	options: {
+		hideControl = true,
+		multiple = false,
+		position = 'right',
+		withSelectionTitle = true
+	} = {},
```
Comment (albinojunior) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/filter/VelocityUnitFilterHooks.es.js]: Define options property with a default value and remove destruction and passing options "{...options}" to Filter component.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/performance-by-assignee-page/PerformanceByAssigneePage.es.js [hunk 6]
```diff
@@ -35,18 +26,17 @@ const PerformanceByAssigneePage = ({query, routeParams}) => {
...
-		filterTitles,
-		filterValues
-	);
+	const filterKeys = ['processStep', 'roles'];
+
+	const {
+		dispatch,
+		filterValues: {roleIds, taskKeys},
+		selectedFilters
+	} = useFilter(filterKeys);
```
Comment (albinojunior) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/performance-by-assignee-page/PerformanceByAssigneePage.es.js]: Did we need to create filterKeys const?

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/workload-by-assignee-page/WorkloadByAssigneePage.es.js [hunk 7]
```diff
@@ -35,18 +26,16 @@ const WorkloadByAssigneePage = ({query, routeParams}) => {
...
-		filterKeys,
-		filterTitles,
-		filterValues
-	);
+	const filterKeys = ['processStep', 'roles'];
+	const {
+		dispatch,
+		filterValues: {roleIds, taskKeys},
+		selectedFilters
+	} = useFilter(filterKeys);
```
Comment (albinojunior) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/workload-by-assignee-page/WorkloadByAssigneePage.es.js]: Did we need to create filterKeys const?

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/shared/components/filter/FilterResultsItem.es.js [hunk 8]
```diff
@@ -51,7 +45,7 @@ class FilterResultsItem extends React.Component {
 							<div className="label-section">
 								<span className="font-weight-normal">{`${filter.name}: `}</span>
 
-								<strong>{itemName}</strong>
+								<strong>{resultName || name}</strong>
```
Comment (albinojunior) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/shared/components/filter/FilterResultsItem.es.js]: resultName does not is a function?

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/shared/components/filter/util/filterUtil.es.js [hunk 9]
```diff
@@ -73,10 +73,28 @@ const getSelectedItems = filterResults => {
...
+			dividerAfter: item.dividerAfter && !!items[index + 1],
+			key
+		};
+	});
+
+const mergeItemsArray = (items1, items2) => {
+	const mergedItems = items1 || [];
+
+	return items2 ? mergedItems.concat(items2) : mergedItems;
+};
```
Comment (albinojunior) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/shared/components/filter/util/filterUtil.es.js]: Can you replace it to:
```javascript
const mergeItemsArray = (items = [], newItems = []) => {
    return items.concat(newItems);
};
```
?

Comment (gabrielpramos) [reply modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/shared/components/filter/util/filterUtil.es.js]: I replaced to another form and made some improvements
`const mergeItemsArray = (baseItems = [], ...items) => {
	items = items.filter(value => value !== undefined && value !== null);
	return baseItems.concat(...items);
};`

**Comment by albinojunior** [comment]
@gabrielpramos resend this PR please.

**Pull request closed without merging**